source/Angle.cpp

```diff
@@ -171,6 +179,17 @@ Point Angle::Rotate(const Point &point) const
 
 
 
+// Judge whether this is inside from "base" to "limit."
+// The range from "base" to "limit" is expressed by "clock" orientation.
+bool Angle::isInRange(const Angle& base, const Angle& limit) const
+{
+	Angle normalizedLimit = limit - base;
+	Angle normalizedTarget = *this - base;
+	return normalizedTarget.angle <= normalizedLimit.angle;
```

Comment:
I'm not sure I understand this code.
And isn't it easier to just perform 2 comparisons `this.angle >= base.angle && this.angle <= limit.angle`?

Reply:
It chooses an edge of the arc as the reference angle (base) and compares relative angles to decide whether this is in the range. It assumes all angles are in 360 degrees from the reference angle.

Reply:
Those calculations are in the range from 0x0 to 0x10000, not in the range from 0 to 360. But your explanation sounds good and this probably also works in the range to 0x10000.